lib/bots/versionbot.ts

```diff
@@ -492,22 +503,60 @@ export class VersionBot extends ProcBot {
                     state: 'failure'
                 },
                 method: this.githubApi.repos.createStatus,
-            }).then(() => {
-                // We only complain on opening. Either at that point it'll be fixed, or we
-                // simply don't pass it in future. This stops the PR being spammed on status changes.
-                if (event.cookedEvent.data.action === 'opened') {
+            });
+        }).then(() => {
+            // Check statuses (including Versionist) on the PR.
+            return this.checkStatuses(pr);
+        }).then((checkStatus) => {
+            // If any of them fail (*not* pending), then *if* we haven't already
+            // commented (ie. we previously commented and there's been no commit
+            // since), we ping the author of the PR and whinge at them.
+            if (checkStatus === StatusChecks.Failed) {
+                // Get the last commit. If there have been no comments *since* the
+                // date of that commit, then we know that we can safely post a
+                // comment telling the author of a failure. If there has, then
+                // it's implicit that we made a comment.
+                const lastCommitTimestamp = lastCommit.commit.committer.date;
+                return this.githubCall({
+                    data: {
+                        owner,
+                        repo: name,
+                        number: pr.number,
+                    },
+                    method: this.githubApi.issues.getComments
+                }).then((comments: GithubApiTypes.Comment[]) => {
+                    // Check for any comments that come *after* the last commit
+                    // timestamp and is a Bot.
+                    let foundBotComment = false;
+                    let warningUsers = '';
+                    for (let comment of comments) {
+                        if ((Date.parse(lastCommitTimestamp) < Date.parse(comment.created_at)) &&
+                            (comment.user.type === 'Bot')) {
+                            foundBotComment = true;
+                            break;
+                        }
+                    }
+                    if (foundBotComment) {
+                        return Promise.resolve();
+                    }
+
+                    // Now we ping the author, telling them something went wrong.
+                    warningUsers = `@${author}, `;
+                    if (author !== committer) {
+                        warningUsers += `@${committer}, `;
+                    }
```

Comment:
Would be nice to refactor this down a little for clarity, to something more like:

```typescript
const lastCommitTimestamp = lastCommit.commit.committer.date;

getBotComments().then((botComments) => {
  if (Date.parse(_.last(botComments).created_at) > Date.parse(lastCommitTimestamp)) {
    let warningUsers = `@${author}, `;
    if (author !== committer) {
      warningUsers += `@${committer}, `;
    }
    commentOnPr(pr, `${warningUsers}status checks...`), 
  }
});
```

Definitely feels like there's quite a few places where we're using `githubCall` directly, which we could wrap up nicely like this to make everything much clearer and more readable.